torchinductor/compile_fx.py

```diff
@@ -102,12 +100,43 @@ def compile_fx_inner(
 
         if len(set(graph.device_types)) > 1:
             log.warning("skipping cudagraphs due to multiple devices")
-        elif graph.mutated_inputs and set(graph.device_types) == {"cuda"}:
-            log.warning("skipping cudagraphs due to input mutation")
-        elif complex_memory_overlap_inputs:
-            log.warning("skipping cudagraphs due to complex input striding")
+        elif set(graph.device_types) == {"cuda"}:
+            if graph.mutated_inputs:
+                log.warning("skipping cudagraphs due to input mutation")
+            elif complex_memory_overlap_inputs:
+                log.warning("skipping cudagraphs due to complex input striding")
 
-    return compiled_fn
+    return align_inputs(compiled_fn, example_inputs, range(num_fixed))
+
+
+def clone_preserve_strides(x):
```

Comment:
if inputs are slices of the same tensor, this blows up the memory, and also removes aliasing but that's pre-existing

Reply:
You think we should switch to recompiling if alignments change? I did this based on your suggestion when we talked in person.

Reply:
Idk, maybe we should handle arg aliasing some other way? It's problematic now anyways (and we are already doing this for cuda graphs) so it's not a regression and definitely not blocking for this PR.